### PR TITLE
Add string benchmarks

### PR DESCRIPTION
**Pull request by ronawho**
Add string versions of argsort, coargsort, groupby, and gather to run as
part of nightly performance testing.

The approach taken here is to create separate `str-<bench>` benchmark
files that just call into the main `<bench>` files with adjusted problem
sizes and number of trials. This makes it easier to fit into the
graphing infrastructure without running all the type variants. See
discussion in https://github.com/mhmerrill/arkouda/pull/640

Resolves https://github.com/mhmerrill/arkouda/issues/356
Resolves https://github.com/mhmerrill/arkouda/issues/534

### CODE REVIEW

**Comment by ronawho** [comment]
Probably easiest to review commit-by-commit. Sample graphs for 16-node-cs-hdr are available [here](https://ronawho.github.io/ak-652/?startdate=2021/02/01&enddate=2021/02/04&graphs=argsortperformance,coargsortperformance,groupbyperformance,gatherperformance,stringargsortperformance,stringcoargsortperformance,stringgroupbyperformance,stringgatherperformance)

File: benchmarks/IO.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 TYPES = ('int64', 'float64')
 
 def time_ak_write_read(N_per_locale, trials, dtype, path, seed):
-    print(">>> arkouda write/read")
+    print(">>> numpy {} write/read".format(dtype))
```
Comment (reuster986) [inline benchmarks/IO.py:11]: This looks like it should be arkouda, rather than numpy...

**Review by reuster986 — APPROVED** [review]
This looks good to me, aside from a minor fix in one of the print statements.

File: benchmarks/IO.py (hunk 1 above)
Comment (ronawho) [reply benchmarks/IO.py]: Whoops, good catch -- must have screwed up a `sed` command. Fixed now

**Comment by reuster986** [comment]
ditto here, pulling master should fix CI

**Review by reuster986 — APPROVED** [review]

**Comment by reuster986** [comment]
@ronawho small merge conflict with the string benchmark PR. Looks like it's just unioning the list of benchmarks to run.

**Comment by ronawho** [comment]
Rebased

**Pull request merged**